### DISCUSSION TITLE
RNode EEPROM map

### DISCUSSION THREAD

**Discussion by faragher** [markqvist/Reticulum#472] [Show and tell]
I've wanted a map of this for personal use, and it was simple enough to puzzle out, but I figured it's worth leaving it here for reference.

On the ESP32, there is no "EEPROM," it simply mapped to flash, however I'm still calling it an EEPROM for interoperability:

Size: 200 bytes
Map:
```
0x00                      Product
0x01                      Model
0x02                      Hardware Revision
0x03 - 0x06 [4 bytes]     Serial Number
0x07 - 0x0a [4 bytes]     Time of creation
0x0b - 0x1a [16 bytes]    Checksum
0x1b - 0x9a [128 bytes]   Signature
0x9b                      Info Lock Byte
0xa7                      Configuration OK
```

Product and Model are defined in ROM.h and the ROM definitions in rnodeconf.py, and subject to change, so they won't be repeated here. The lock byte appears to always be 0x73. Everything else is straightforward, and appears to be big-endian (ie: 00 00 00 01 is 1, not 16 million+)

The signature appears to not matter for RNode functionality, but is for authentication of firmware if you're dumping the firmware to other repos/machines.

**Comment by Erethon** [comment]
Thank you for this, do you think it might be best to move this to the Wiki maybe to draw more attention to it? See #473.

Regardless, I want to add that the `Checksum` is an MD5 hash of [Product, Model, Hardware Revision, Serial Number, Time of Creation] and that there's also the `Conf OK Byte` at location 0xA7 which is set to 0x73 (same as Info Lock Byte).
